cpp/tests/fixed_point/fixed_point_tests.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <vector>
+#include <algorithm> // transform, generate_n
+#include <numeric>   // iota, accumulate
+#include <limits>
+
+struct FixedPointTest : public cudf::test::BaseFixture {};
+
+using namespace cudf::fp;
+using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
```

Comment:
If `fixed_point` is intended to be used with multiple `Rep` types, please add tests for those as well.

Reply:
It is intended, but I think that will be done in a follow up job when we extend to Decimal64. For now I believe we are focusing on Decimal32. Correct me if I am wrong @jrhemstad - happy to add the tests now if people would like

Reply:
With the SFINAE constructor, lgtm for now.

Reply:
I seem to remember the initial request is for Decimal64 (?).